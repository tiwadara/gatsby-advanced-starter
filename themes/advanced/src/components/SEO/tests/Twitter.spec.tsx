import TwitterTags from "../Twitter";
import {
  tagListHasEmptyValues,
  tagListHasUniqueKeys,
  sampleSeoData,
} from "./TestUtils";

describe("seo module TwitterTags", () => {
  it("generates correct tags on website pages", () => {
    expect.assertions(1);

    const generatedTags = TwitterTags({
      seoData: sampleSeoData.seoWebsite,
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags on article pages", () => {
    expect.assertions(1);

    const generatedTags = TwitterTags({
      seoData: sampleSeoData.seoArticle,
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags when missing description", () => {
    expect.assertions(1);

    const generatedTags = TwitterTags({
      seoData: { ...sampleSeoData.seoArticle, description: undefined },
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags when missing imageUrl", () => {
    expect.assertions(1);

    const generatedTags = TwitterTags({
      seoData: { ...sampleSeoData.seoArticle, imageUrl: undefined },
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("doesn't generate empty tags", () => {
    expect.assertions(6);

    // Test for website pages
    const websiteTags = TwitterTags({
      seoData: sampleSeoData.seoWebsite,
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(tagListHasEmptyValues(websiteTags)).toBe(false);

    // Test for article pages
    const articleTags = TwitterTags({
      seoData: sampleSeoData.seoArticle,
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(tagListHasEmptyValues(articleTags)).toBe(false);

    // Test for missing userTwitterName
    const tagsWithoutUserTwitter = TwitterTags({
      seoData: sampleSeoData.seoArticle,
      userData: { ...sampleSeoData.user, twitterName: undefined },
      websiteData: sampleSeoData.website,
    });

    expect(tagListHasEmptyValues(tagsWithoutUserTwitter)).toBe(false);

    // Test for missing siteTwitterName
    const tagsWithoutSiteTwitterName = TwitterTags({
      seoData: sampleSeoData.seoArticle,
      userData: sampleSeoData.user,
      websiteData: { ...sampleSeoData.website, twitterName: undefined },
    });

    expect(tagListHasEmptyValues(tagsWithoutSiteTwitterName)).toBe(false);

    // Test for missing description
    const tagsWithoutDescription = TwitterTags({
      seoData: { ...sampleSeoData.seoArticle, description: undefined },
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(tagListHasEmptyValues(tagsWithoutDescription)).toBe(false);

    // Test for missing imageUrl
    const tagsWithoutImageUrl = TwitterTags({
      seoData: { ...sampleSeoData.seoArticle, imageUrl: undefined },
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(tagListHasEmptyValues(tagsWithoutImageUrl)).toBe(false);
  });

  it("generates unique keys", () => {
    expect.assertions(2);

    // Test for website pages
    const generatedWebsiteTags = TwitterTags({
      seoData: sampleSeoData.seoWebsite,
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(tagListHasUniqueKeys(generatedWebsiteTags)).toBe(true);

    // Test for article pages
    const generatedArticleTags = TwitterTags({
      seoData: sampleSeoData.seoArticle,
      userData: sampleSeoData.user,
      websiteData: sampleSeoData.website,
    });

    expect(tagListHasUniqueKeys(generatedArticleTags)).toBe(true);
  });
});
