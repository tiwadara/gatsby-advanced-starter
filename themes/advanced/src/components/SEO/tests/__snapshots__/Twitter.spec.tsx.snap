// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seo module TwitterTags generates correct tags on article pages 1`] = `
Array [
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Post title"
    name="twitter:title"
  />,
  <meta
    content="Post description."
    name="twitter:description"
  />,
  <meta
    content="/logos/logo-1024.png"
    name="twitter:image"
  />,
  <meta
    content="Post cover image alt text."
    name="twitter:image:alt"
  />,
  <meta
    content="Vagr9K"
    name="twitter:creator"
  />,
  <meta
    content="Vagr9K"
    name="twitter:site"
  />,
]
`;

exports[`seo module TwitterTags generates correct tags on website pages 1`] = `
Array [
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Website title"
    name="twitter:title"
  />,
  <meta
    content="Website description."
    name="twitter:description"
  />,
  <meta
    content="/logos/logo-1024.png"
    name="twitter:image"
  />,
  <meta
    content="Website image alt description."
    name="twitter:image:alt"
  />,
  <meta
    content="Vagr9K"
    name="twitter:creator"
  />,
  <meta
    content="Vagr9K"
    name="twitter:site"
  />,
]
`;

exports[`seo module TwitterTags generates correct tags when missing description 1`] = `
Array [
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Post title"
    name="twitter:title"
  />,
  <meta
    content="/logos/logo-1024.png"
    name="twitter:image"
  />,
  <meta
    content="Post cover image alt text."
    name="twitter:image:alt"
  />,
  <meta
    content="Vagr9K"
    name="twitter:creator"
  />,
  <meta
    content="Vagr9K"
    name="twitter:site"
  />,
]
`;

exports[`seo module TwitterTags generates correct tags when missing imageUrl 1`] = `
Array [
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Post title"
    name="twitter:title"
  />,
  <meta
    content="Post description."
    name="twitter:description"
  />,
  <meta
    content="Post cover image alt text."
    name="twitter:image:alt"
  />,
  <meta
    content="Vagr9K"
    name="twitter:creator"
  />,
  <meta
    content="Vagr9K"
    name="twitter:site"
  />,
]
`;
